tests/core/data/test_network.py

```diff
@@ -0,0 +1,334 @@
+import numpy as np
+from numpy.testing import assert_almost_equal
+import pytest
+
+from qutip.core import data
+from qutip.core.data import dense
+from qutip_tensornetwork.core.data import Network
+from qutip_tensornetwork.testing import assert_network_close
+from .conftest import random_node, random_complex_network
+import tensornetwork as tn
+
+def test_init_only_in_edge():
+    node = random_node((2,))
+    network = Network([], node[0:], copy=False)
+    assert len(network.nodes) == 1
+    assert network.nodes.pop() is node
```

Comment:
```suggestion
    assert network.nodes[0] is node
```
Modify the `.nodes` while checking them seems a bit odd, so let's just do `.nodes[0]` instead if we can.

Reply:
unfortunately sets are not subscriptable. This is another reason I was considering to store nodes as lists.

Reply:
I removed the pop as much as I could but I left it for those cases where I thought changing it for `list(network.nodes)[0]` was not worth.